src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
+  }
+
+  public inefficientSort(arr: number[]): number[] {
+    return arr.sort((a, b) => {
+      if (a > b) return 1;
+      if (a < b) return -1;
+      return 0;
+    });
+  }
```

Comment:
PullProofAI

The `inefficientSort` method uses the `sort` function directly on the input array, which sorts the array in place and mutates the original array. This can lead to unexpected behavior as the original array gets modified. To maintain immutability and prevent side effects, consider creating a copy of the input array before sorting it.

```diff
+ return [...arr].sort((a, b) => {
```



<!-- This is an auto-generated comment by PullProofAI -->